docs/source/common/test_set.rst

```diff
@@ -1,17 +1,38 @@
 .. _test_set:
 
-Test set
-========
-Lightning forces the user to run the test set separately to make sure it isn't evaluated by mistake.
-Testing is performed using the ``trainer`` object's ``.test()`` method.
+##########
+Evaluation
+##########
+
+During and after training we need a way to evaluate our models to make sure they are not overfitting while training and
+generalize well on unseen or real-world data. There are generally 2 stages of evaluation: validation and testing. To some
+degree they serve the same purpose, to make sure models works on real data but they have some practical differences.
+
+Validation is usually done during training, traditionally after each training epoch. It can be used for hyperparameter optimization or tracking model performance during training.
+It's a part of the training process.
+
+Testing is usually done once we are satisfied with the training and use the best model selected from the validation metrics and want to evaluate its performance on unseen data where the model acts
+as a black box. The test dataset is emulating the production/benchmark data where labels aren't available.
```

Comment:
```suggestion
Testing is usually done once we are satisfied with the training and only with the best model selected from the validation metrics.
```

There's a lot of nuance here in reality so I would keep these explanations as short as possible. We want to motivate the functionality, not teach ML here